Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift

```diff
@@ -96,15 +96,15 @@ class SentrySessionReplay: NSObject {
         delegate?.sessionReplayStarted(replayId: sessionReplayId)
     }
 
-    func pause() {
+    func pauseSessionMode() {
```

Comment:
just curious: would it be possible to drop this method entirely, and just call `pause()` when there's no internet connection? We'll stop getting display link callbacks anyway and not capture new frames, right?

Reply:
just thinking that we have `isSessionPaused` variable, and then we set it to `false` in the `pause` method which might look confusing

Reply:
- The pause function stops everything.
- `pauseSessionMode` keeps taking screenshots when replay is in buffer mode to at least capture the first error.

I dont think we can remove `pauseSessionMode`, but I need to reevaluate if we should disable `isSessionPaused` during pause. This needs to be in another PR tho.

Reply:
> pauseSessionMode keeps taking screenshots when replay is in buffer mode to at least capture the first error.

does it keep taking screenshots or am I misinterpreting this line? https://github.com/getsentry/sentry-cocoa/blob/e1c38196160220f9d7580685802a1b2a5b580c29/Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift#L181

Reply:
Yes, it keeps taking screenshots. But only for buffer mode